nemo/collections/tts/torch/data.py

```diff
@@ -471,16 +505,44 @@ def __getitem__(self, index):
         if sample["is_phoneme"] == 1:
             rel_audio_path_as_text_id += "_phoneme"
 
-        # Load audio
-        features = self.featurizer.process(
-            sample["audio_filepath"],
-            trim=self.trim,
-            trim_ref=self.trim_ref,
-            trim_top_db=self.trim_top_db,
-            trim_frame_length=self.trim_frame_length,
-            trim_hop_length=self.trim_hop_length,
-        )
-        audio, audio_length = features, torch.tensor(features.shape[0]).long()
+        if (
+            self.segment_max_duration is not None
+            and 'duration' in sample
+            and sample['duration'] > self.segment_max_duration
+        ):
```

Comment:
Out of curiosity--why do we need the separate handling for this now? If the featurizer doesn't handle max duration, should it do that instead of having that logic here?

Reply:
Added a comment about segment max duration argument in docstring. This portion is used only for speaker verification task in SSLDisentangler where we need segmented audios. If audio length is greater than segment_max_duration, we take a random segment of segment_max_duration.